servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/KeepAliveManager.java

```diff
@@ -343,13 +352,16 @@ private void gracefulCloseWriteSecondGoAway() {
         gracefulCloseState = State.GRACEFUL_CLOSE_SECOND_GO_AWAY_SENT;
 
         channel.writeAndFlush(new DefaultHttp2GoAwayFrame(NO_ERROR)).addListener(future -> {
-            if (activeStreams == 0) {
-                close0();
+            if (!future.isSuccess()) {
+                LOGGER.debug("{} Failed writing the second GO_AWAY", channel, future.cause());
+                close0(future.cause());
```

Comment:
Consider mentioning in the log that the cannel is now going to be proactively closed.